### PR TITLE
Improve graphql usage

### PR DESCRIPTION
**Pull request by WickyNilliams**
**DO NOT MERGE UNTIL SCHEMA MERGED**

### What is the context of this PR?
Backwards-compatible changes to use `ID` type for entity id's, and input types for all mutations

### How to review 
run tests, examine changes, test against master of `eq-author` and [associated PR](https://github.com/ONSdigital/eq-author/pull/211) (to ensure it is compatible with both)

### CODE REVIEW

**Comment by WickyNilliams** [comment]
[Improve GraphQL usage (5)](https://trello.com/c/8bkFSzmj/187-improve-graphql-usage-5)

File: docker-compose.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ services:
       - db
     volumes:
       - .:/app
-#      - ./node_modules/eq-author-graphql-schema:/app/node_modules/eq-author-graphql-schema
+      - ./node_modules/eq-author-graphql-schema:/app/node_modules/eq-author-graphql-schema
```
Comment (samiwel) [inline docker-compose.yml]: Is it right that we want to uncomment this?

File: schema/resolvers.js [hunk 2]
```diff
@@ -1,21 +1,31 @@
-const { merge, includes } = require("lodash");
-const GraphQLDate = require("graphql-iso-date").GraphQLDate;
+const { GraphQLDate } = require("graphql-iso-date");
+const { includes } = require("lodash");
+
+const getId = args => args.newId || args.id;
+const getNewId = entity => entity.id.toString(10);
+const getInputArgs = args => args.input || args;
+const getIdFromArgs = args => getInputArgs(args).id;
```
Comment (samiwel) [inline schema/resolvers.js]: Would there ever be a need to check for `newId` here? e.g. `const getIdFromArgs = args => getId(getInputArgs(args));`

Comment (WickyNilliams) [reply schema/resolvers.js]: No, should be fine. This is only used for mutations, there the named arg is called `id` and the field on the input type is also `id`

File: docker-compose.yml (hunk 1 above)
Comment (WickyNilliams) [reply docker-compose.yml]: Good catch

**Pull request merged**